parcels/particlefile/particlefilenodes.py

```diff
@@ -0,0 +1,229 @@
+                     while ParticleFile is given as an argument of ParticleSet.execute()
+                     It is either a timedelta object or a positive double.
+    :param write_ondelete: Boolean to write particle data only when they are deleted. Default is False
+    :param convert_at_end: Boolean to convert npy files to netcdf at end of run. Default is True
+    :param tempwritedir: directories to write temporary files to during executing.
+                     Default is out-XXXXXX where Xs are random capitals. Files for individual
+                     processors are written to subdirectories 0, 1, 2 etc under tempwritedir
+    :param pset_info: dictionary of info on the ParticleSet, stored in tempwritedir/XX/pset_info.npy,
+                     used to create NetCDF file from npy-files.
+    """
+    max_index_written = -1
+
+    def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, convert_at_end=True,
+                 tempwritedir=None, pset_info=None):
+        """
+        ParticleFileNodes - Constructor
+        :param name: Basename of the output file
+        :param particleset: ParticleSet to output
+        :param outputdt: Interval which dictates the update frequency of file output
+                         while ParticleFile is given as an argument of ParticleSet.execute()
+                         It is either a timedelta object or a positive double.
+        :param write_ondelete: Boolean to write particle data only when they are deleted. Default is False
+        :param convert_at_end: Boolean to convert npy files to netcdf at end of run. Default is True
+        :param tempwritedir: directories to write temporary files to during executing.
+                         Default is out-XXXXXX where Xs are random capitals. Files for individual
+                         processors are written to subdirectories 0, 1, 2 etc under tempwritedir
+        :param pset_info: dictionary of info on the ParticleSet, stored in tempwritedir/XX/pset_info.npy,
+                         used to create NetCDF file from npy-files.
+        """
+        super(ParticleFileNodes, self).__init__(name=name, particleset=particleset, outputdt=outputdt,
+                                                write_ondelete=write_ondelete, convert_at_end=convert_at_end,
+                                                tempwritedir=tempwritedir, pset_info=pset_info)
+        self.var_names.append('index')
+        self.max_index_written = 0
+        self.time_written = []
+
+    def __del__(self):
+        """
+        ParticleFileNodes - Destructor
+        """
+        super(ParticleFileNodes, self).__del__()
+
+    def _reserved_var_names(self):
+        """
+        :returns the reserved dimension names not to be written just once.
+        """
+        return ['time', 'lat', 'lon', 'depth', 'id', 'index']
+
+    def _create_trajectory_records(self, coords):
```

Comment:
well, this additional variable is entirely key for the writer to work - just because here, the NetCDF file writer need to convert the IDs to (local) indices, while the global index is represented in this additional, written `index` variable.

Now, with a second look, yes, perhaps `_create_trajectory_records(...)` could _perhaps_ inherit from the baseparticleset.py. I can try it out locally, see if it works.

Reply:
And, did it work?

Reply:
I adapted that a bit, and it works now in a revised format